mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotation.java

```diff
@@ -0,0 +1,128 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.model;
+
+import java.util.Map;
+
+import org.apache.james.mailbox.exception.UnsupportedAnnotationException;
+import org.apache.james.mailbox.model.MailboxACL.MailboxACLEntryKey;
+
+/**
+ * Stores an Metadata Extensioin applicable to a mailbox. Inspired by
+ * RFC5464 The IMAP METADATA Extension.
+ * 
+ * Implementations must be immutable. Implementations should override
+ * {@link #hashCode()} and {@link #equals(Object)}.
+ * 
+ */
+public interface MailboxAnnotation {
+    /**
+     * The key used in {@link MailboxAnnotation#getEntries()}. Implementations should
+     * override {@link #hashCode()} and {@link #equals(Object)} in such a way
+     * that all of {@link #getName()}, {@link #getNameType()} and
+     * {@link #isNegative()} are significant.
+     * 
+     */
+    interface MailboxAnnotationEntryKey {
```

Comment:
I think MailboxAnnotation can be a POJO with just two fields : 
 - a String for the key
 - a String for the value